racket/src/cs/main.sps

```diff
@@ -929,6 +929,10 @@
          (when (and n (exact-nonnegative-integer? n))
            (set-schedule-quantum! n)))))
 
+   (let* ([config (read-installation-configuration-table)]
```

Comment:
I think this turns out not to be a valid place to call `read-installation-configuration-table`. When I build and run `racketcs` as created in `racket/build/cs/c`, for example, a `build-path` error gets printed as a result of a bad caching of the config directory. The error doesn't happen for the installed `racket`, but that's probably because caching and finding the config directory is complicated.

Similar to the reordering that happens in BC, maybe the best thing here in CS is to have `initialize-place!` return the config table that it reads, and then use that table to extract `'build-stamp` (a few lines below where the banner printout currently happens, inside the `call-on-main-thread` callback). That has the additional benefit of not reading the configuration file twice. I don't think it will be a problem to delay banner output that way.

Reply:
Ah okay, I wasn't expecting `racket/src/build/cs/c/racketcs` and `racket/bin/racket` to have different behaviour here, but indeed they do. Thanks for the tip, I'm slowly understanding how all these startup paths work together. 🙂 

I've moved the config table portion inside `initialize-place!`, but I decided to set the build stamp there (instead of returning the config table) so that all places would have the build stamp value. It's probably quite unlikely that someone will call `banner` from an additional place, but if they do, it seems best for it to have the same value in all places.

Reply:
Hm... All places share the same instance of Rumble (and any other layer implemented by a ".sls"), so initializing the banner stamp in `initialize-place!` does not sound right.

But I see that some other things in `initialize-place!` also set shared items: `set-host-collects-dir!`, `set-host-config-dir!`, `set-collects-dir!`, and `set-config-dir!`. In fact, probably that was the source of the trouble calling `read-installation-configuration-table` before `initialize-place!`.

Maybe the right thing is to merge your change, because resetting the variables will be harmless, but then I can look at cleaning up these initialization steps.

Reply:
Ah okay! I indeed assumed that initialisation of shared items was needed for each place based on those other `set...!` calls. 😅 

Merging this as-is (matching the current redundant initialisation of place variables) and then separately cleaning up initialisation seems like a good strategy to me. 🙂